biodatasets/pubmed_qa/pubmed_qa.py

```diff
@@ -0,0 +1,245 @@
+
+_DESCRIPTION = """\
+PubMedQA is a novel biomedical question answering (QA) dataset collected from PubMed abstracts.
+The task of PubMedQA is to answer research biomedical questions with yes/no/maybe using the corresponding abstracts. 
+PubMedQA has 1k expert-annotated, 61.2k unlabeled and 211.3k artificially generated QA instances. 
+
+Each PubMedQA instance is composed of:
+  (1) a question which is either an existing research article title or derived from one, 
+  (2) a context which is the corresponding abstract without its conclusion, 
+  (3) a long answer, which is the conclusion of the abstract and, presumably, answers the research question, and
+  (4) a yes/no/maybe answer which summarizes the conclusion.
+
+PubMedQA is the first QA dataset where reasoning over biomedical research texts, 
+especially their quantitative contents, is required to answer the questions.
+"""
+
+_HOMEPAGE = "https://github.com/pubmedqa/pubmedqa"
+_LICENSE = "MIT License"
+
+_URLS = {
+    "pqaa": "https://drive.google.com/uc?export=download&id=1kaU0ECRbVkrfjBAKtVsPCRF6qXSouoq9",
+    "pqal": "https://drive.google.com/uc?export=download&id=1kQnjowPHOcxETvYko7DRG9wE7217BQrD",
+    "pqau": "https://drive.google.com/uc?export=download&id=1q4T_nhhj8UvJ9JbZedhkTZHN6ZeEZ2H9",
+}
+
+_SUPPORTED_TASKS = [Tasks.QUESTION_ANSWERING]
+_SOURCE_VERSION = "1.0.0"
+_BIGBIO_VERSION = "1.0.0"
+
+_CLASS_NAMES = [
+    "Yes",
+    "No",
+    "Maybe"
+]
+
+class PubmedQADataset(datasets.GeneratorBasedBuilder):
+    """PubmedQA Dataset"""
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    BIGBIO_VERSION = datasets.Version(_BIGBIO_VERSION)
+
+    BUILDER_CONFIGS = [
+        # PQAU & PQAA Source
+        BigBioConfig(
+            name="pqaa_source",
+            version=SOURCE_VERSION,
+            description="PubmedQA artificial source schema",
+            schema="source",
+            subset_id="pqaa",
+        ),
+        BigBioConfig(
+            name="pqau_source",
+            version=SOURCE_VERSION,
+            description="PubmedQA unlabeled source schema",
+            schema="source",
+            subset_id="pqau",
+        ),
+        # PQAU & PQAA BigBio Schema
+        BigBioConfig(
+            name="pqaa_bigbio_qa",
+            version=BIGBIO_VERSION,
+            description="PubmedQA artificial BigBio schema",
+            schema="bigbio_qa",
+            subset_id="pqaa",
+        ),
+        BigBioConfig(
+            name="pqau_bigbio_qa",
+            version=BIGBIO_VERSION,
+            description="PubmedQA unlabeled BigBio schema",
+            schema="bigbio_qa",
+            subset_id="pqau",
+        ),
+    ] + [
+        # PQAL Source Schema        
+        BigBioConfig(
+            name=f"pqal_fold{i}_source",
+            version=datasets.Version(_SOURCE_VERSION),
+            description="PubmedQA labeled source schema",
+            schema="source",
+            subset_id=f"pqal_fold{i}"
+        ) for i in range(10)
+    ] + [
```

Comment:
it seems that these cv splits are defined but don't change how the data is loaded below.

Reply:
@galtay : it will change the folder path on the `_split_generators` function for PQA-L by defining referring the folder using the subset_id.
https://github.com/bigscience-workshop/biomedical/blob/4852e8bbff8fd8b77fb583ab09ea14917f3e9645/biodatasets/pubmed_qa/pubmed_qa.py#L166-L179

Reply:
Oh, maybe it is because most of the train data on different folds are the same given a certain index.
I checked `fold0`, `fold1`, and `fold2`on the validation set and it seems to work as expected.

<img width="1095" alt="Screenshot 2022-04-15 at 11 16 35 AM" src="https://user-images.githubusercontent.com/2826602/163512697-34592995-7776-409c-ae4d-5396c77219a6.png">

Reply:
yes, I think that is what was confusing me!